### ISSUE TITLE
spirv-fuzz: Add barrier instructions in dead blocks

### ISSUE THREAD

**Issue by afd** [KhronosGroup/SPIRV-Tools#4420]
Barrier instructions must be placed with care, because they lead to undefined behavior if executed under non-uniform control flow.

However, we can place them arbitrarily in dead blocks.

**Comment by afd** [comment]
@Mostafa-ashraf19 This would be a good one for you to work on in due course.

**Comment by paulthomson** [comment]
Yes, sounds good. I suggest starting with memory barrier instructions.
